src/aliceVision/hdr/hdrTestCommon.hpp

```diff
@@ -0,0 +1,178 @@
+#pragma once
+#include <aliceVision/image/all.hpp>
+#include <aliceVision/system/Logger.hpp>
+
+#include "rgbCurve.hpp"
+#include "sampling.hpp"
+
+#include <boost/filesystem.hpp>
+#include <random>
+#include <array>
+
+namespace aliceVision {
```

Comment:
It would be good to keep hdr as for all elements in this module:
aliceVision::hdr::test

Reply:
This would become problematic soon due to namespace ambiguities. Consider this situation:
 - There is `aliceVision::hdr::test` and `aliceVision::otherModule::test` namespaces 
 - Some code does `using namespace aliceVision::hdr` and `using namespace aliceVision::otherModule`.
 - It's now impossible to call anything from `test::` as `test::ABC` even if `ABC` is unique. It's enough for `test` to be ambiguous between `aliceVision::hdr::test` and `aliceVision::otherModule::test`.

Due to this reason it's an antipattern to sprinkle commonly used names like `test` around the codebase.

I think it's unlikely that there will ever be any ambiguity between names within `hdrTestCommon.hpp`, so I think that given the above limitations `aliceVision::test` is not really bad name.

Reply:
But if you insist I will of course put code into `aliceVision::hdr::test`, just tell me that you don't think the above situation is likely to happen.

Reply:
I see your point, but for the sake of simplicity I would like to keep using the module name explicitly in the namespace:
if the file is in "aliceVision/something", the namespace is "aliceVision::something".

For redundant names, I agree, but in all case, if you put everything in aliceVision::test you will have the same conflicts.
With the first option, you will always be able to use the full name to disambiguate if needed. In all case, this discussion is theoretical, as these tests elements will only be used in the corresponding test, so there is no reason that we will use multiple ones.

So better to keep the simple naming rule everywhere.

Reply:
OK, I will put the code into `aliceVision::hdr::test`.

> if you put everything in aliceVision::test you will have the same conflicts.

Just for the record, this is not the case. A more clearer example that fails to compile:

```
namespace A::test 
{
    int fun1();
}

namespace B::test 
{
    int fun2();
}

int abc()
{
    using namespace A;
    using namespace B;
    return test::fun1();
}
```

Reply:
The namespace issue has been fixed now.